### DISCUSSION TITLE
误修改某临时文件导致参考文献不能编译

### DISCUSSION THREAD

**Discussion by qiaoxiaoyue** [sjtug/SJTUThesis#774] [提问] [answered]
原本是编译成功的，中途修改了bib文件，未编译点击了保存，后再编译出现参考文献无法编译的问题，显示Citation undifined。
于是通过源代码管理放弃了某些临时文件的修改，且将bib文件回归原样，但现在依然无法编译参考文献，Citation undifined，
主文件显示错误
Please (re)run Biber on the file:
(biblatex)	main
(biblatex)	and rerun LaTeX afterwards.
请问要怎么修复？
马上就写完论文了，突然这样子，挺急的，求救，谢谢了

Mac OS系统
使用Vs code

**Comment by skyzh** [comment, marked as answer]
latexmk -C

**Reply by qiaoxiaoyue** [reply]
能再具体点讲讲吗？我现在编译空白论文模板也是同样的问题，应该是不小心修改了latex编译时的某文件，或者配置的问题？

**Reply by skyzh** [reply]
我们的 Makefile 和 Compile.bat 都有清理所有临时文件的功能。运行下面任意一句：

```
make clean
./Compile.bat clean
latexmk -C
```

即可。

**Reply by skyzh** [reply]
如果是更改了 TeXLive 发行版里的内容，那就没有办法了，只能重装了……但看你的问题描述应该没有改到这些文件

**Reply by qiaoxiaoyue** [reply]
试了清理，当下是不行的。然后重启大法好，解决了。
不过还有一点挺奇怪的，重新下载的模板还是无法编译参考文献。
十分感谢！

**Comment by LogCreative** [comment]
无法编译参考文献的话 考虑删除 biber 的本地缓存
```shell
rm -rfv $(biber --cache)
```
可能需要提供更多的日志

https://github.com/plk/biber/issues/312
